hazelcast/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -182,6 +182,19 @@ default boolean closeIsCooperative() {
     default void close(@Nullable Throwable error) throws Exception {
     }
 
+    /**
+     * Returns {@code true} if this instance is stateful.
+     * <p>
+     * When a job is to be submitted, the job definition ({@link DAG} or
+     * {@link Pipeline}) is serialized. This serialization can be avoided if the
+     * local member is the job coordinator and the {@link DAG} is stateless.
+     *
+     * @since 5.3
+     */
+    default boolean isStateful() {
+        return false;
+    }
```

Comment:
The bug was found in v5.2.1 and the fix will be backported to 5.2.x once we are done with reviewing.

Reply:
Yes, so for this reason I think the default value should be `true`. Otherwise, this won't fix the regression, it will only allow for a future fix where we find a case where the PMS is stateful.